pkgs/flox-activation-scripts/default.nix

```diff
@@ -1,84 +1,106 @@
   editorconfig = builtins.path {
     name = "editorconfig";
     path = ../../.editorconfig;
   };
+
 in
-runCommand "flox-activation-scripts"
+runCommandNoCC "flox-activation-scripts"
   {
-    buildInputs = [
-      bash
-      coreutils
-      gnused
-      daemonize
+    nativeBuildInputs = [ gnused ];
+    outputs = [
+      "out"
+      "wrapper"
     ];
   }
   ''
-    cp -R ${../../assets/activation-scripts} $out
+    # Create the "out" output.
+    cp -R ${activation-scripts} $out
+    chmod +x $out/activate
+    chmod -R +w $out
+    patchShebangs $out/activate
+    substituteInPlace $out/activate --replace-fail "__OUT__" "$out"
+
+    # Next create the (lesser) "wrapper" output.
+    cp -R ${activation-scripts} $wrapper
+    chmod +x $wrapper/activate
+    chmod -R +w $wrapper
+    patchShebangs $wrapper/activate
+    substituteInPlace $wrapper/activate --replace-fail "__OUT__" "$wrapper"
 
-    substituteInPlace $out/activate \
-      --replace "@bash@" "${bash}/bin/bash" \
-      --replace "@coreutils@" "${coreutils}" \
-      --replace "@flox_activations@" "${flox-activations}/bin/flox-activations" \
-      --replace "@getopt@" "${getopt}" \
-      --replace "@gnused@" "${gnused}" \
-      --replace "@jq@" "${jq}/bin/jq" \
-      --replace "@nawk@" "${nawk}" \
-      --replace "@out@" "$out" \
-      --replace "@process-compose@" "${process-compose}/bin/process-compose" \
-      --replace "@setsid@" "${util-linux}/bin/setsid" \
-      --replace "@daemonize@" "${daemonize}/bin/daemonize" \
-      --replace "/usr/bin/env bash" "${bash}/bin/bash"
+    # TODO: come up with neater way to master activate script for wrapper case.
 
-    substituteInPlace $out/activate.d/bash \
-      --replace "@gnused@" "${gnused}"
-    substituteInPlace $out/activate.d/fish \
-      --replace "@gnused@" "${gnused}"
-    substituteInPlace $out/activate.d/tcsh \
-      --replace "@gnused@" "${gnused}"
-    substituteInPlace $out/activate.d/zsh \
-      --replace "@gnused@" "${gnused}"
+    # Remove start-services.bash.
+    rm $wrapper/activate.d/start-services.bash
+    sed -i 's/source ".*start-services.bash"/: no services in wrapper script/' $wrapper/activate
 
-    for i in $out/etc/profile.d/*; do
-      substituteInPlace $i --replace "@coreutils@" "${coreutils}"
-      substituteInPlace $i --replace "@gnused@" "${gnused}"
-      substituteInPlace $i --replace "@findutils@" "${findutils}"
-      substituteInPlace $i --replace "@ld-floxlib@" "${ld-floxlib_so}"
-      substituteInPlace $i --replace "@ldconfig@" "${ldconfig}"
-      substituteInPlace $i --replace "@nawk@" "${nawk}"
-      substituteInPlace $i --replace "@fd@" "${fd}"
-    done
+    # Remove references to flox-activations.
+    rm $wrapper/activate.d/attach-*.bash
+    sed -i 's/source ".*attach-.*.bash"/: no attaching in wrapper script/' $wrapper/activate
+    sed -i 's/_flox_activations=.*/_flox_activations=true/' \
+        $wrapper/activate $wrapper/activate.d/start.bash
```

Comment:
^ I'm not a huge fan of replacing the imports by regex, id rather do two separate substitutions and then copy them to the outputs separately

```
cp -R ${activation-scripts-user-mode} $out
cp -R ${activation-scripts-manifest-build-wrapper} $wrapper
```

Reply:
Great question - most definitely yes, by not having them in the first place!  😄 

Please consider all of the above as temporary while we work to further refactor this logic. What we actually need is a _modular_ design in which simply removing a file will _in a single stroke_ both disable the functionality _and_ remove that module's package requirements from the closure. For instance, if we had instead implemented the attachment logic as files in `etc/profile.d` then this could have been possible, so that's the next investigation I'd suggest we perform.